rs/ic_os/manifest_tool/src/components_parser.rs

```diff
@@ -0,0 +1,127 @@
+use anyhow::{Context, Result};
+use regex::Regex;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+pub const COMPONENTS_PATH: &str = "ic-os/components";
+
+#[derive(Debug)]
+pub struct IcosManifest {
+    pub setupos: Manifest,
+    pub hostos: Manifest,
+    pub guestos: Manifest,
+    pub boundary_guestos: Manifest,
+}
+
+impl IcosManifest {
+    pub fn new(
+        guestos_components: Manifest,
+        hostos_components: Manifest,
+        setupos_components: Manifest,
+        boundary_guestos_components: Manifest,
+    ) -> Self {
+        IcosManifest {
```

Comment:
I wonder if this tool needs to be rigid to our component structure, of if it could be a list that keeps paths to give context when needed. I'm thinking of that test down below where we have to create empty files, but I don't think these should be optional either 🤔.

I'm not sold either way 😅.

Reply:
Yeah, I was thinking that when we restructure our components format (which may be soon if David moves forward with the [sh_binary](https://bazel.build/reference/be/shell#sh_binary) update) we would have to update this, but I think we should wait for the component structure to change. Then we'll have a clearer picture to work with.